src/stream/src/executor/backfill/cdc/upstream_table/snapshot.rs

```diff
@@ -131,9 +131,9 @@ impl UpstreamTableRead for UpstreamTableReader<ExternalStorageTable> {
         #[for_await]
         for chunk in chunk_stream {
             let chunk = chunk?;
-            let cardinality = chunk.cardinality();
+            let chunk_size = chunk.capacity();
```

Comment:
I'm not sure whether we should limit the `size` or `visible size` in fact. In some code we also do computation for invisible rows, e.g. HashAgg.

Reply:
> I'm not sure whether we should limit the size or visible size in fact. In some code we also do computation for invisible rows, e.g. HashAgg.

Then let's leave the behaviour untouched for now.

Reply:
But when being built by `StreamChunkBuilder`, the chunk size counted both visible and invisible rows, and since the `max_chunk_size` given to `StreamChunkBuilder` is `min(config.chunk_size, rate_limit)`, I guess it's reasonable to use the same counting here. I don't have strong opinion as well, need discussion.

Reply:
IC, StreamChunkBuilder splits records in such a way that includes considers invisible rows. And that is decided by rate limit. Likewise rate limiting should then consider invisible rows to be consistent as well. Okay that's reasonable to me.

I was pushing back mainly because it is an out of scope change that's independent of the bugfix.
The rationale makes sense to me, just prefer this kind of change to be in a separate PR.